src/transformers/models/phi/configuration_phi.py

```diff
@@ -0,0 +1,175 @@
+            The dropout ratio after computing the attention scores.
+        hidden_act (`str` or `function`, *optional*, defaults to `"gelu_new"`):
+            The non-linear activation function (function or string) in the decoder.
+        max_position_embeddings (`int`, *optional*, defaults to 2048):
+            The maximum sequence length that this model might ever be used with. Phi-1 and Phi-1.5 supports up to 2048
+            tokens.
+        initializer_range (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        layer_norm_eps (`float`, *optional*, defaults to 1e-05):
+            The epsilon used by the rms normalization layers.
+        use_cache (`bool`, *optional*, defaults to `True`):
+            Whether or not the model should return the last key/values attentions (not used by all models). Only
+            relevant if `config.is_decoder=True`. Whether to tie weight embeddings or not.
+        tie_word_embeddings (`bool`, *optional*, defaults to `False`):
+            Whether to tie weight embeddings
+        rope_theta (`float`, *optional*, defaults to 10000.0):
+            The base period of the RoPE embeddings.
+        rope_scaling (`Dict`, *optional*):
+            Dictionary containing the scaling configuration for the RoPE embeddings. Currently supports two scaling
+            strategies: linear and dynamic. Their scaling factor must be an float greater than 1. The expected format
+            is `{"type": strategy name, "factor": scaling factor}`. When using this flag, don't update
+            `max_position_embeddings` to the expected new maximum. See the following thread for more information on how
+            these scaling strategies behave:
+            https://www.reddit.com/r/LocalPersimmon/comments/14mrgpr/dynamically_scaled_rope_further_increases/. This
+            is an experimental feature, subject to breaking API changes in future versions.
+        partial_rotary_factor (`float`, *optional*, defaults to 0.5):
+            Percentage of the query and keys which will have rotary embedding.
+        qk_layernorm (`bool`, *optional*, defaults to `False`):
+            Whether or not to normalize the Queries and Keys after projecting the hidden states
+        bos_token_id (`int`, *optional*, defaults to 1):
+            Denotes beginning of sequences token id.
+        eos_token_id (`int`, *optional*, defaults to 2):
+            Denotes end of sequences token id.
+
+    Example:
+
+    ```python
+    >>> from transformers import PhiModel, PhiConfig
+
+    >>> # Initializing a Phi-1 style configuration
+    >>> configuration = PhiConfig.from_pretrained("susnato/phi-1_dev")
+
+    >>> # Initializing a model from the configuration
+    >>> model = PhiModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+    model_type = "phi"
```

Comment:
```suggestion
    model_type = "phi"
```
We are gonna have the same problem as with Faclon, the online model type is `"model_type": "mixformer-sequential",`.
For this, I think we'll have to either keep it or find a workaround! (Otherwise TGI could break)
but that will be the last thing to solve!

Reply:
Can't we just change the `model_type` to `phi` when we update the weights there?

Reply:
No let's just use the one they have online that will be the simplest!

Reply:
Hi @ArthurZucker, I changed the model_type from `"phi"` to `"mixformer-sequential"` but I am getting the `test_load_with_mismatched_shapes` error, I tried to change the mappings in `models/auto` but still getting the error...

Reply:
I have changed `model_type = "phi" -> "mixformer-sequential"` and added a line in the test modelling file `test_mismatched_shapes = False` which skips the failed test for now. 
Later after we transfer the weights and update the online config we can revert back to `model_type = "phi"` and remove the line.

WDYT?

Reply:
No the error is here to tell you that you need to update the `config_mapin` meaning you need to make sure that `configuration_auto` also has the correct name! 😉 
Let's add a test to make sure that if we load a model from the current revision online with this code , the generated outputs are correct! (using the code_revision and revision flags!)

Reply:
Hey I did change the `CONFIG_MAPPING_NAMES` from `("phi", "PhiConfig")->("mixformer-sequential", "PhiConfig")` But the error was still there :), 
Is there anything more I have to change?

Reply:
Also we can't load either phi-1 or phi-1.5 from the current online version because the weights(namings) won't match :(

Reply:
Please see [this comment](https://github.com/huggingface/transformers/pull/26170#issuecomment-1778720835).